src/sidebar/Sidebar.js

```diff
@@ -38,6 +39,24 @@ class Sidebar extends Adw.NavigationPage {
   #initializeSidebar() {
     this.browse_view = new BrowseView(this);
     this.search_view = new SearchView();
+    this.flattened_model = this.#newListStore();
+
+    let doc_index = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+      "doc-index.json",
+    );
+
+    const content = doc_index.load_contents(null);
```

Comment:
I am loading/reading the JSON file synchronously here because when I do it asynchronously, the app launches quickly but the sidebar is blank for a second before it instantly gets populated with all the items.

Instead, I'm loading it synchronously which makes up for a slightly longer startup time but when the app does launch, its sidebar is completely populated and the webview is loaded as well. Unfortunately, I couldn't find an easy way to just asynchronously stream the JSON file data one element at time and so we will have to load the entire JSON file at once.